### PR TITLE
cbarn support

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-616.mdx [hunk 1]
```diff
@@ -14,42 +14,95 @@ author: Óscar Garries
...
 	}
-	int sol = INT_MAX;
-	for (int start = 0; start < n; start++) {
-		int cur = 0, cur_cows = cows;
-		for (int i = 0; i < n; i++) {
-			cur_cows -= a[(start + i) % n];
-			cur += cur_cows;
+
+	int min_dist = INT32_MAX;
+	for (int start = 0; start < room_num; start++) {
```
Comment (TruaShamu) [inline solutions/usaco-616.mdx:33]:
```suggestion
	for (int unlock = 0; unlock < room_num; unlock++) {
```
can be changed to match the java and py code

File: solutions/usaco-616.mdx [hunk 2]
```diff
@@ -14,42 +14,95 @@ author: Óscar Garries
...
+		int dist = 0;
+		int cows_left = total_cows;
+		for (int r = 0; r < room_num; r++) {
+			cows_left -= rooms[(start + r) % room_num];
```
Comment (TruaShamu) [inline solutions/usaco-616.mdx:37]:
```suggestion
			cows_left -= rooms[(unlock+ r) % room_num];
```
here as well

**Review by TruaShamu — APPROVED** [review]
looks good to me

File: solutions/usaco-616.mdx [hunk 3]
```diff
@@ -14,42 +14,95 @@ author: Óscar Garries
...
+```
+
+</JavaSection>
+<PySection>
+
+```py
+with open('cbarn.in') as read:
+	room_num = int(read.readline())
+	rooms = [int(read.readline()) for _ in range(room_num)]
+	assert len(rooms) == room_num
```
Comment (ryanchou-dev) [inline solutions/usaco-616.mdx:91]: unnecessary

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/usaco-616.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-616.mdx:91]: @TruaShamu if you agree i'll delete

Comment (TruaShamu) [reply solutions/usaco-616.mdx:91]: sure, go ahead

**Pull request merged**